google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractReadContext.java

```diff
@@ -190,8 +190,12 @@ private SingleReadContext(Builder builder) {
       this.bound = builder.bound;
       // single use transaction have a single RPC and hence there is no need
       // of a channel hint. GAX will automatically choose a hint when used
-      // with a multiplexed session.
-      this.channelHint = getChannelHintOptions(session.getOptions(), null);
+      // with a multiplexed session. But GAX currently uses an AtomicLong to perform round-robin
+      // channel selection and that will cause contention issues at high-concurrency. So we are
+      // passing a random hint instead of doing GAX round-robin.
```

Comment:
Please remove this/replace this comment. Gax uses an `AtomicInteger` to implement round-robin, and that does not introduce any contention.